cookbooks/aws-parallelcluster-byos/resources/create_user.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at http://aws.amazon.com/apache2.0/
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+resource_name :create_user
+provides :create_user
+
+# Resource to create system user
+
+property :name, String, name_property: true
+property :system_users, Array, required: false
+
+default_action :run
+
+def check_gid(gid)
+  check_group_stdout = Mixlib::ShellOut.new("getent group #{gid}").run_command.stdout.strip
+  return unless check_group_stdout.empty?
```

Comment:
you can check the `cmd.error?` here and look at the stdout only if needed later

Reply:
done